pkgs/development/python-modules/matplotx/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pytestCheckHook
+, matplotlib
+, numpy
+, networkx
+, pypng
+, scipy
+}:
+
+buildPythonPackage rec {
+  pname = "matplotx";
+  version = "0.3.10";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "nschloe";
+    repo = "matplotx";
+    rev = "v${version}";
+    hash = "sha256-EWEiEY23uFwd/vgWVLCH/buUmgRqz1rqqlJEdXINYMg=";
+  };
+
+  propagatedBuildInputs = [
+    matplotlib
+    numpy
+    networkx
+    pypng
+    scipy
```

Comment:
These packages are optional dependencies so they should be added to `passthru.optional-dependencies`.
See, [17.27.2.3.2. Optional extra dependencies](https://nixos.org/manual/nixpkgs/stable/#reference)